worker/provisioner/provisioner_task.go

```diff
@@ -1046,16 +1032,16 @@ func (task *provisionerTask) startMachine(
 		// If the failure of StartInstance() is due the availability zone,
 		// find a new one to use when retrying.
 		if err == environs.ErrAvailabilityZoneFailed {
-			if placementZone {
+			if strings.Contains(startInstanceParams.Placement, "zone") {
 				// User specified availability zone failed, nothing to retry.
 				return task.setErrorStatusAndRemoveMachineAZ("cannot start instance for machine %q in placement zone: %v", machine, err)
 			}
-			azRemaining, err2 := task.markMachineFailedInAZ(machine, startInstanceParams.Placement)
+			azRemaining, err2 := task.markMachineFailedInAZ(machine, startInstanceParams.AvailabilityZone)
 			if err2 != nil {
-				return err2
+				return task.setErrorStatusAndRemoveMachineAZ("cannot start instance: %v", machine, err2)
```

Comment:
an error here would be unexpected, I think? in which case we should be returning it up the stack, causing the provisioner to restart. so...

```go
if err := task.setErrorStatus("cannot start instance: %v", machine, err); err != nil {
    logger.Errorf("setting error status: %s", err)
}
return err2
```

Reply:
it's being returned up the stack now, the provisioner will restart.